src/profiles/profiles.rs

```diff
@@ -0,0 +1,519 @@
+        if self.when.is_empty() {
+            return Ok(true);
+        }
+        for c in self.when.iter() {
+            if !c.matches()? {
+                return Ok(false);
+            }
+        }
+        Ok(true)
+    }
+}
+
+fn default_name() -> String {
+    "Default".to_string()
+}
+
+/// Profile information.
+#[derive(Deserialize, Debug)]
+#[serde(rename_all = "snake_case")]
+pub(crate) struct Profile {
+    /// Name of the profile.
+    pub(crate) name: String,
+    #[allow(dead_code)]
+    /// Version of the Profile API that is being used.
+    pub(crate) version: Version,
+    /// Information about the profile in human readable format.
+    pub(crate) about: Option<String>,
+    /// Collect Profiles
+    #[serde(default = "Vec::new")]
+    pub(crate) collect: Vec<CollectProfile>,
+}
+
+impl Profile {
+    /// Find a profile
+    pub fn find(name: &str) -> Result<Profile> {
+        for path in get_profile_paths()?.iter().filter(|p| p.as_path().exists()) {
+            for entry in path.read_dir()? {
+                let entry = entry?;
+                match Profile::load(entry.path()) {
+                    Ok(mut profiles) => {
+                        for profile in profiles.drain(..) {
+                            if profile.name.eq(name) {
+                                return Ok(profile);
+                            }
+                        }
+                    }
+
+                    Err(err) => {
+                        debug!("Skipping invalid file {}: {err}", entry.path().display())
+                    }
+                }
+            }
+        }
+        bail!("Profile with name {name} not found");
+    }
+
+    /// Load a profile from a path.
+    /// A file can contain multiple yaml objects so we return a list of objects.
+    pub fn load(path: PathBuf) -> Result<Vec<Profile>> {
+        let mut result = Vec::new();
+        let contents = read_to_string(path)?;
+        for document in serde_yaml::Deserializer::from_str(&contents) {
+            result.push(Profile::deserialize(document)?);
+        }
+        Ok(result)
+    }
+
+    /// Load a profile from a string.
+    #[cfg(test)]
+    pub fn from_str(contents: &str) -> Result<Profile> {
+        Ok(serde_yaml::from_str(contents)?)
+    }
+
+    /// Evaluate collect profiles and return the one that matches.
+    pub fn match_collect(&self) -> Result<Option<&CollectProfile>> {
+        if self.collect.is_empty() {
+            return Ok(None);
+        }
+
+        for p in self.collect.iter() {
+            if p.matches()? {
+                return Ok(Some(p));
+            }
+        }
+        Ok(None)
+    }
+
+    /// Generate cli arguments from a profile. The result is a list of arguments that can be
+    /// concatenated to the ones provided by the user.
+    pub fn cli_args(&self, subcommand: &str) -> Result<Vec<OsString>> {
+        let mut result = Vec::new();
+        let (args, flags) = match subcommand {
```

Comment:
it was essentially because of yaml syntax reasons. Maybe the null YAML can work, I'll give it a try. Having "true" does not work because some flags don't allow a value.

Reply:
We cannot just discard it because some options do need it :-D But it seems the null character works!